### PR TITLE
feat: Add monitor dockerization and remove private build

### PR DESCRIPTION
**Pull request by gitferry**
This PR is to dockerize the monitor. Also, this PR removes the private build of the reporter and the submitter as the major Babylon repos are no longer private.

### CODE REVIEW

File: contrib/images/monitor/wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	echo "The binary $(basename "${BINARY}") cannot be found. Please add the binary to the shared folder. Please use the BINARY environment variable if the name of the binary is not 'vigilante'"
+	exit 1
+fi
+
+export BABYLONGENESIS="/babylon/config/genesis.json"
+export VIGILANTECONFIG="/vigilante/${CONFIG}"
+export MONITORLOG="/vigilante/${LOG}"
+
+if [ -d "$(dirname "${REPORTERLOG}")" ]; then
+  "${BINARY}" reporter --config "${VIGILANTECONFIG}" --genesis "${BABYLONGENESIS}" 2>&1 | tee  "${REPORTERLOG}"
```
Comment (SebastianElvis) [inline contrib/images/monitor/wrapper.sh:19]: Here all `reporter`s should be `monitor` right?

```suggestion
  "${BINARY}" monitor --config "${VIGILANTECONFIG}" --genesis "${BABYLONGENESIS}" 2>&1 | tee  "${REPORTERLOG}"
```

File: contrib/images/monitor/wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+else
+  "${BINARY}" reporter --config "${VIGILANTECONFIG}" --genesis "${BABYLONGENESIS}" 2>&1
```
Comment (SebastianElvis) [inline contrib/images/monitor/wrapper.sh:21]:
```suggestion
  "${BINARY}" monitor --config "${VIGILANTECONFIG}" --genesis "${BABYLONGENESIS}" 2>&1
```

File: contrib/images/monitor/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+## Image for building
+FROM golang:1.19-alpine AS build-env
+
+RUN apk add --no-cache openssh-client git make build-base linux-headers libc-dev \
+                       pkgconfig zeromq-dev musl-dev alpine-sdk libsodium-dev \
+                       libzmq-static libsodium-static
+
+ENV GO111MODULE=on
+# Download public key for github.com
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
```
Comment (SebastianElvis) [inline contrib/images/monitor/Dockerfile:10]: Is this still needed?

File: contrib/images/reporter/Dockerfile [hunk 4]
```diff
@@ -1,19 +1,13 @@
...
-ARG pass
-
 RUN apk add --no-cache openssh-client git make build-base linux-headers libc-dev \
                        pkgconfig zeromq-dev musl-dev alpine-sdk libsodium-dev \
                        libzmq-static libsodium-static
 
 ENV GO111MODULE=on
-ENV GOPRIVATE=github.com/babylonchain/babylon
 # Download public key for github.com
 RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
```
Comment (SebastianElvis) [inline contrib/images/reporter/Dockerfile:10]: Is this still needed?

File: contrib/images/submitter/Dockerfile [hunk 5]
```diff
@@ -1,19 +1,13 @@
...
-ARG pass
-
 RUN apk add --no-cache openssh-client git make build-base linux-headers libc-dev \
                        pkgconfig zeromq-dev musl-dev alpine-sdk libsodium-dev \
                        libzmq-static libsodium-static
 
 ENV GO111MODULE=on
-ENV GOPRIVATE=github.com/babylonchain/babylon
 # Download public key for github.com
 RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
```
Comment (SebastianElvis) [inline contrib/images/submitter/Dockerfile:10]: ditto

**Review by SebastianElvis — APPROVED** [review]
Lgtm! just some nits and minor questions

File: contrib/images/monitor/wrapper.sh (hunk 1 above)
Comment (gitferry) [reply contrib/images/monitor/wrapper.sh:19]: Nice catch!

File: contrib/images/monitor/Dockerfile (hunk 3 above)
Comment (gitferry) [reply contrib/images/monitor/Dockerfile]: Just tested it and we don't need it. Thanks!

**Review by vitsalis — APPROVED** [review]

**Pull request merged**